### ISSUE TITLE
The use of different oxidation states leads to different Coulomb energy rankings.

### ISSUE THREAD

**Issue by BigBMonster** [orex/supercell#68]
First of all, thank you for developing this program.

This is my structure:
[Li3NiMnFe0.75Cu0.25O6.txt](https://github.com/user-attachments/files/17620528/Li3NiMnFe0.75Cu0.25O6.txt)

I used Cu to substitute Fe.

I used the second method mentioned in #67, namely the average oxidation state method, to calculate the Coulomb energy.

I calculated the Coulomb energy using different oxidation states:

First attempt: Cu: +1, Fe: 3.666666
Second attempt: Cu: +2, Fe: 3.333333

This is the command I used for the first run:
./supercell -i LiCoO211.cif -s 2x2x1 -p "Ni*:p=4" -p "Fe*:p=3" -p "Mn*:p=4" -p "Cu*:p=1" -p "Ni*:c=2" -p "Fe*:c=3.66666666" -p "Mn*:c=4" -p "Cu*:c=1" -p "Li*:c=1" -p "O*:c=-2" -q -m -n l100 -o cell_out6/cell > cell_out6/log.out

Here is the Coulomb energy calculated:
[energy1.txt](https://github.com/user-attachments/files/17607306/energy1.txt)

This is the command used for the second run:
./supercell -i LiCoO211.cif -s 2x2x1 -p "Ni*:p=4" -p "Fe*:p=3" -p "Mn*:p=4" -p "Cu*:p=1" -p "Ni*:c=2" -p "Fe*:c=3.33333333" -p "Mn*:c=4" -p "Cu*:c=2" -p "Li*:c=1" -p "O*:c=-2" -q -m -n l100 -o cell_out7/cell > cell_out7/log.out

Here is the Coulomb energy calculated:
[energy2.txt](https://github.com/user-attachments/files/17607307/energy2.txt)

My question is:

(1) I noticed that the ranking of the Coulomb energies changes depending on the oxidation states used. I’d like to understand where the issue might lie.Which result should I trust?

**Comment by Gitdowski** [comment]
TL;DR -> see the last paragraph

It depends on what you define as "trust". Rigorously speaking, I'd say, you should not trust any of the outputted structures. This is because supercell simply calculates the Coulomb energy of a system of point charges using the specified positions from the cif file and the given oxidation states you provided. 

If you want to trust supercell completely, then you 
a) need to trust your provided oxidation states completely 
b) need to trust your provided atomic positions completely 
c) need to trust that any of the quantum mechanical effects that are covered by a DFT calculation can be neglected and that the Coulomb energy perfectly describes the energy of the system
By the way, if all these three points were true, it would not be necessary to do any DFT calculation at all for your system.

Of course we know that the real system is not a distribution of point charges, but positively charged cores with electron clouds surrounding them. If you start thinking how oxidation states can be defined, you will see that this is actually not as straight forward as your high school chemistry course taught you, see for example https://mattermodeling.stackexchange.com/questions/1439/what-are-the-types-of-charge-analysis . Then we know that positions will change during a relaxation with DFT, and that these relaxations can depend on the local distribution of atoms. That is why I say: You cannot trust supercell 100% because I would not trust the given input parameters (positions and oxidation states) to 100%.

So, what's then the point of using supercell at all? You will see, that even in a small system, there can be millions of ways to arrange the atoms, and all these structures cannot be treated with DFT. Therefore, we need to find a way, or in other words a simplified model, to identify a few promising structures. If I understand it correctly, for you this means low energy structures. For this purpose, a simple energy ordering based on Coulomb energies might be helpful as a first order approximation. But as you can see in your first energy file, the first 100 structures differ by roughly 1 eV. With your input structure of 12 atoms and a 2x2x1 cell of it this makes 48 atoms, so roughly 20 meV/atom. I'd say this is an energy range on which the "DFT effects" matter. So I would at least put the first 10 structures into DFT and see which one gives the lowest energy.

Lastly I want to mention again that the Coulomb energy of course depends on the provided oxidation states. We cannot tell you whether your first attempt or your second attempt is the better one to describe your system realistically. I proposed using an averaged oxidation state as a measure to simplify the supercell model. Nobody stops you to define a second set of Fe atoms. You just need to give them a different name. Any species you do not have in your structure works, e.g. Xe to have something completely different, feel free to use anything else. Supercell does not care about the name, it only needs to know the oxidation states you want these species to have. If you that by copying the Fe entries in your cif file and rename them to Xe, then you can give your first set of Fe atoms a +3 oxidation state and the second one a +4 state: 
-p "Fe*:c=3" -p "Xe*:c=4" 
Of course you need to take care again, that the overall charge state is always zero by defining the appropriate number of Fe and Xe atoms then or by adding the appropriate occupation numbers in the cif file. Before you put such a structure into DFT, you will need to rename the Xe back to Fe.


My recommendation if you do not know what charge states to expect for Fe and Cu a priori: Why don't you try out the first ten structures from both attempts, relax them in DFT and see which of them gives the lowest energy and if you can identify the oxidation states?

**Comment by Gitdowski** [comment]
As a last comment to your 

> I’d like to understand where the issue might lie.

There is no issue.

Supercell relies on a model that uses your provided point charges to calculate the electrostatic energy. You provide different input, you get different output.  Whether your input makes sense is a different question **you need to answer/understand**.

**Comment by BigBMonster** [comment]
Thank you very much for your detailed response; it has been very helpful to me. I wish you a wonderful life.
